ProcessQueue/__init__.py

```diff
@@ -30,76 +31,119 @@ def process_scan_report_sheet_table(sheet):
+    max_column = sheet.max_column
     # Set a column index (openpyxl index at 1 instead of 0)
-    col_idx=1
-    
-    for row_idx,row_cell in enumerate(sheet.iter_rows(min_row=1,max_col=max_column),start = 1):
-  
-    # Works through pairs of value/frequency columns
-        for column_idx,cell in enumerate(row_cell):
+
+    for row_idx, row_cell in enumerate(
+        sheet.iter_rows(min_row=1, max_col=max_column), start=1
+    ):
+
+        # Works through pairs of value/frequency columns
+        for column_idx, cell in enumerate(row_cell):
+            freq = sheet.cell(row=row_idx, column=column_idx + 2).value
             if cell.value:
                 if (column_idx) % 2 == 0:
-                # As we move down rows, checks that there's data there
-                # This is required b/c value/frequency col pairs differ
-                # in the number of rows
-                  if sheet.cell(row=row_idx,column=column_idx+1).value == "" and sheet.cell(row=row_idx,column=column_idx+2).value == "":
-                    continue
-                # Append to Results as (Field Name,Value,Frequency)
-                  results.append(
-                      (sheet.cell(row=1,column=column_idx+1).value,sheet.cell(row=row_idx,column=column_idx+1).value, sheet.cell(row=row_idx,column=column_idx+2).value)
-                  )
-    col_idx=col_idx+1
+                    # As we move down rows, checks that there's data there
+                    # This is required b/c value/frequency col pairs differ
+                    # in the number of rows
+                    if (
+                        sheet.cell(row=row_idx, column=column_idx + 1).value == ""
+                        and sheet.cell(row=row_idx, column=column_idx + 2).value == ""
+                    ):
+                        continue
+
+                    # Append to Results as (Field Name,Value,Frequency)
+                    results.append(
+                        (
+                            sheet.cell(row=1, column=column_idx + 1).value,
+                            sheet.cell(row=row_idx, column=column_idx + 1).value,
+                            sheet.cell(row=row_idx, column=column_idx + 2).value,
+                        )
+                    )
+            else:
+                # If only frequency is present add to results
+                if freq:
+                    if (column_idx) % 2 == 0:
+                        results.append(
+                            (
+                                sheet.cell(row=1, column=column_idx + 1).value,
+                                cell.value,
+                                sheet.cell(row=row_idx, column=column_idx + 2).value,
+                            )
+                        )
 
     return results
 
 
+def get_field_id(dict, field):
+    """
+    Input: dictionary with field_id:field_value pairs
+    Input: field name
+    Returns: field id
+    """
+    for id, name in dict.items():

```

Comment:
The more I think about this the more I think it can be simplified. As it stands, we create `id_x_names` as a dictionary of `id:name`. We then use `get_field_id()` to do a reverse-lookup: given a name, return the id associated to it. And this just returns the first matching name, because we assume that the name will be unique. This is the only place `id_x_names` is used.

If that's the case, then we might as well create a reverse dictionary `name:id` and just do a lookup. If need be, we can add in a guard to flag any time when it tries to add a second value to the same `name`. Then we can get rid of this whole `get_field_id()` function.

Reply:
Yeah that makes sense.This used to be name:id but I switched it around because names weren't always unique. So since it's no longer an issue I'll change it back to that and remove the get_field_id()